src/connectors/apple-music.js

```diff
@@ -1,22 +1,29 @@
 'use strict';
 
-const ARTISTALBUM_SEPARATOR = '—';
+const shadowRoot = document.querySelector('amp-lcd').shadowRoot;
 
-const artistAlbumSelector = '.web-chrome-playback-lcd__sub-copy-scroll-inner-text-wrapper';
+const queryShadowRoot = (selector) => shadowRoot.querySelector(selector);
 
-Connector.playerSelector = '.web-chrome';
+Connector.getArtist = () => queryShadowRoot('.lcd-meta__secondary .lcd-meta-line__fragment:first-child').textContent;
 
-Connector.currentTimeSelector = '#apple-music-current-playback-time';
+Connector.getAlbum = () => queryShadowRoot('.lcd-meta__secondary .lcd-meta-line__fragment:last-child').textContent;
 
-Connector.remainingTimeSelector = '#apple-music-current-playback-time-remaining';
+Connector.getCurrentTime = () => queryShadowRoot('.lcd-progress__time--elapsed').textContent;
 
-Connector.trackSelector = '.web-chrome-playback-lcd__song-name-scroll-inner-text-wrapper';
+Connector.getRemainingTime = () => queryShadowRoot('.lcd-progress__time--remaining').textContent;
 
-Connector.getTrackInfo = () => {
-	const artistAlbum = Util.getTextFromSelectors(artistAlbumSelector);
-	return Util.splitArtistAlbum(artistAlbum, [ARTISTALBUM_SEPARATOR]);
-};
+Connector.getTrack = () => queryShadowRoot('.lcd-meta__primary-wrapper .lcd-meta-line__fragment').textContent;
 
-Connector.isPlaying = () => {
-	return $(Connector.playerSelector).hasClass('is-playing');
-};
+Connector.isPlaying = () => document.querySelector('.playback-play__pause').getAttribute('aria-hidden') !== 'true';
```

Comment:
Would the following work here, to prevent an error when the page first loads?
`Util.getAttrFromSelectors('.playback-play__pause', 'aria-hidden') !== 'true';`

Reply:
fixed, nice catch!

Reply:
Nice work on this. I had similar ideas to get it working and attempted updating the connector yesterday but quickly ran out of patience before I could get anything usable to submit. 😅
BTW I wonder if each occurrence of `textContent` needs additional checks, as the following error also appears on page load:
`Uncaught TypeError: Cannot read properties of null (reading 'textContent')`
But everything seems to work once playback is initiated.

Reply:
yep, it indeed doesnt really affect anything, it just throws a little error when the element doesn't exist (which is when you dont want to scrobble anyway). Thanks to javascript being javascript this doesn't really affect anything else.
Regardless, I just pushed another commit with some optional chaining that eliminates the errors, changes no behavior afaik.

Reply:
Nice concise solution! 😎